### PR TITLE
Add legend based on current layer

### LINKED ISSUE #53: Legends for map layers
-Only one layer will ever be on the map at once
-Some layers are choropleth
-Some layers are graduated symbols, would need color ramp in legend as well as circle size if possible

### PR DESCRIPTION
**Pull request by mmcfarland**
## Overview

Uses same configuration for layer symbology to define legend breaks.
Circles are based on the _y15 value set, and the high/low values are
taken from the breaks developed from the entire feature layer for that
value.

Fixes #53 

### Demo

screenshot from 2017-11-09 13 32 56

screenshot from 2017-11-09 13 33 11

### Notes
* Some styles aren't set, and in that case the legend doesn't display any values.  As I'm anticipating those values being added, I didn't address that edge case here.
* I think there is an existing bug with tracts not showing when switching between different MSAs, the legend doesn't update for the same reason.  I think fixing the layers will fix the legend.
* The legends are quite large - that's all I can say.  @designmatty, any ideas for improvements?

## Testing
When loading national vs tract level layers and variables, the legend should appear and match the symbology except for where the bugs listed above prevent it.

### CODE REVIEW

**Comment by designmatty** [comment]
I can cleanup after merged

File: app/assets/js/src/main.js [hunk 1]
```diff
@@ -262,6 +267,40 @@ $(document).ready(function() {
...
+                    layerName: _.findWhere(TCVIZ.Config.nationwide_layers, { value: layerId}).text,
+                });
+
+            $('#legend-inner').html(TCVIZ.Templates.legend(breakCtx));
+            if (TCVIZ.State.sizeBreaks) {
+                $('#legend-circle-inner').html(TCVIZ.Templates.legendCircle(circleCtx));
+            }
+        } else {
+            var ctx = _.assign(layerConfig, {
+                layerName: _.findWhere(TCVIZ.Config.MSA_layers, { value: layerId}).text,
```
Comment (CloudNiner) [inline app/assets/js/src/main.js]: Just a heads up, my chart styling PR snuck a function in that gets the MSA layer label when passed the id `msaLabelForId(msaId)`. Could make it a bit more generic to handle both MSA and nationwide layers and use it throughout your legend code.

**Comment by mmcfarland** [comment]
Fixed the merge conflict.  Will have a look at that function once the in-process column name/config mismatch is resolved (can't load sites atm).

**Comment by mmcfarland** [comment]
@CloudNiner Added 2772e45 to make the label stuff a little easier on the eyes, per your suggestion.  I also rebased on `master` so you'll to wipe your previous branch if you pull this again.

**Comment by mmcfarland** [comment]
Just noticed this bug on my branch, will add another commit to fix.

screenshot from 2017-11-09 15 49 57

**Comment by mmcfarland** [comment]
Fixed in 801f465

**Comment by CloudNiner** [comment]
Nice. Still LGTM!

**Pull request merged**